### PR TITLE
Fix hint on mean(::Number, ::Number)

### LINKED ISSUE #99: Bad error message on mean(::Int, ::Int)
```
julia> using Statistics
julia> mean(1,2)
ERROR: MethodError: objects of type Int64 are not callable
Maybe you forgot to use an operator such as *, ^, %, / etc. ?
Stacktrace:
 [1] mean(f::Int64, itr::Int64)
   @ Statistics /Applications/Julia-1.7.1.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Statistics/src/Statistics.jl:69
 [2] top-level scope
   @ REPL[4]:1
```

### PR DESCRIPTION
**Pull request by LilithHafner**
Back before I know about `middle`, I tried `mean(1, 2)` and was briefly confused before settling on `mean((1, 2))` or `mean([1, 2])`. A better hint in the error message would have helped me.

Before:
```julia
julia> mean(1, 2)
ERROR: MethodError: objects of type Int64 are not callable
Maybe you forgot to use an operator such as *, ^, %, / etc. ?
Stacktrace:
[...]
```
After:
```julia
julia> mean(1, 2)
ERROR: ArgumentError: mean(f, itr) requires a function and an iterable.
       Perhaps you meant middle(x, y)?
Stacktrace:
[...]
```

Fixes #99

### CODE REVIEW

**Review by mschauer — APPROVED** [review]

**Comment by LilithHafner** [comment]
This adds more code just for an error message, so I can see why folks might be hesitant to merge it and/or decide to decline this PR. Is that what is happening, or has this simply fallen by the wayside?

**Comment by LilithHafner** [comment]
Bump

File: src/Statistics.jl [hunk 1]
```diff
@@ -103,6 +103,16 @@ julia> mean(√, [1 2 3; 4 5 6], dims=2)
 """
 mean(f, A::AbstractArray; dims=:) = _mean(f, A, dims)
 
+function mean(f::Number, itr::Number)
+    f_value = try
+        f(itr)
+    catch MethodError
+        rethrow(ArgumentError("""mean(f, itr) requires a function and an iterable.
+                                 Perhaps you meant middle(x, y)?""",))
```
Comment (nalimilan) [inline src/Statistics.jl:111]: How about not putting a newline in the error message?
```suggestion
        rethrow(ArgumentError("mean(f, itr) requires a function and an iterable. " *
                              "Perhaps you meant middle(x, y)?"))
```

**Review by nalimilan — COMMENTED** [review]
Sorry, it's just that nobody took care of it. Sounds reasonable given that it's been reported at https://github.com/JuliaStats/Statistics.jl/pull/143 too.

BTW, a similar problem exists for `sum` in Base.

File: src/Statistics.jl (hunk 1 above)
Comment (LilithHafner) [reply src/Statistics.jl:111]: I was following the pattern of
```
julia> (4+1)(5)
ERROR: MethodError: objects of type Int64 are not callable
Maybe you forgot to use an operator such as *, ^, %, / etc. ?
Stacktrace:
 [1] top-level scope
   @ REPL[9421]:1
```
Putting the hint on a different line as the error, but either way works.

Comment (nalimilan) [reply src/Statistics.jl:111]: Ah OK. I'm not sure whether there are strong conventions in that regard.

**Pull request merged**